lib/FileSystemInfo.js

```diff
@@ -1211,7 +1215,12 @@ class FileSystemInfo {
 						break;
 					}
 					case RBDT_FILE_DEPENDENCIES: {
-						// TODO this probably doesn't work correctly with ESM dependencies
+						// Check for known files without dependencies
+						if (/\.json5?$|\.yarn-integrity$|yarn\.lock$|\.ya?ml/.test(path)) {
```

Comment:
What about markdown?

Reply:
That's usually not imported in node.js so it doesn't matter that much, but I also thought about that and concluded that it's not totally safe to do since there are dialects like `MDX` that could have dependencies.